src/main/java/uk/gov/hmcts/divorce/caseworker/service/updater/MiniApplication.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.divorce.caseworker.service.updater;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.type.Document;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.common.updater.CaseDataContext;
+import uk.gov.hmcts.divorce.common.updater.CaseDataUpdater;
+import uk.gov.hmcts.divorce.common.updater.CaseDataUpdaterChain;
+import uk.gov.hmcts.divorce.document.DocAssemblyService;
+import uk.gov.hmcts.divorce.document.DocumentIdProvider;
+import uk.gov.hmcts.divorce.document.content.DocmosisTemplateProvider;
+import uk.gov.hmcts.divorce.document.content.DraftApplicationTemplateContent;
+import uk.gov.hmcts.divorce.document.model.DivorceDocument;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.divorce.document.DocumentConstants.DIVORCE_MINI_APPLICATION;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.DIVORCE_MINI_APPLICATION_DOCUMENT_NAME;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.DIVORCE_APPLICATION;
+
+@Component
+@Slf4j
+public class MiniApplication implements CaseDataUpdater {
+
+    @Autowired
+    private DocAssemblyService docAssemblyService;
+
+    @Autowired
+    private DocmosisTemplateProvider docmosisTemplateProvider;
+
+    @Autowired
+    private DocumentIdProvider documentIdProvider;
+
+    @Autowired
+    private DraftApplicationTemplateContent draftApplicationTemplateContent;
+
+    @Override
+    public CaseDataContext updateCaseData(final CaseDataContext caseDataContext,
+                                          final CaseDataUpdaterChain caseDataUpdaterChain) {
+
+        log.info("Executing handler for generating mini application for case id {} ", caseDataContext.getCaseId());
+
+        final var updatedCaseData = caseDataContext.copyOfCaseData();
+
+        final var templateName = docmosisTemplateProvider.templateNameFor(
+            DIVORCE_MINI_APPLICATION,
+            updatedCaseData.getApplicant1().getLanguagePreference());
+
+        final Map<String, Object> templateData = draftApplicationTemplateContent.apply(
+            updatedCaseData,
+            caseDataContext.getCaseId(),
+            caseDataContext.getCreatedDate());
+
+        final var documentInfo = docAssemblyService.renderDocument(
+            templateData,
+            caseDataContext.getCaseId(),
+            caseDataContext.getUserAuthToken(),
+            templateName,
+            DIVORCE_MINI_APPLICATION_DOCUMENT_NAME
+        );
+
+        final var ccdDocument = new Document(
+            documentInfo.getUrl(),
+            documentInfo.getFilename(),
+            documentInfo.getBinaryUrl()
+        );
+
+        final var divorceDocument = DivorceDocument
```

Comment:
Seems like a few hoops to jump through here converting the return type of `docAssemblyService.renderDocument` to a `DivorceDocument`. Maybe we should make a utility function for that.